src/protocol/primitives/nullable-string.ts

```diff
@@ -0,0 +1,26 @@
+import type { ReadBuffer } from '../serialization/read-buffer';
+import { Deserializable, type Serializable } from '../serialization/serialization';
+import type { WriteBuffer } from '../serialization/write-buffer';
+
+export class NullableString extends Deserializable implements Serializable {
+  constructor(public readonly value: string | null) {
+    super();
+  }
+
+  public static deserialize(buffer: ReadBuffer): NullableString {
+    const length = buffer.readUInt(2);
```

Comment:
It is specified as Int16 instead of UInt16:

> Represents a sequence of characters or null. For non-null strings, first the length N is given as an INT16. Then N bytes follow which are the UTF-8 encoding of the character sequence. A null value is encoded with length of -1 and there are no following bytes.